### ISSUE TITLE
diff() can produce ops in the wrong order

### ISSUE THREAD

**Issue by asayers** [idubrov/json-patch#6]
There are two possible patches which will take you from `[1,2]` to `[]`:
```
[{"op":"remove","path":"/1"},{"op":"remove","path":"/0"}]
[{"op":"remove","path":"/0"},{"op":"remove","path":"/0"}]
```
The `diff()` function gives the following (invalid) patch:
```
[{"op":"remove","path":"/0"},{"op":"remove","path":"/1"}]
```

**Comment by idubrov** [comment]
🙀

**Comment by idubrov** [comment]
Fixed in 0.2.4. Fix is a bit sketchy, but seems to work 😁

**Comment by asayers** [comment]
Thanks!
